### PR TITLE
Edit/Delete Contributions on Activity Page

### PR DESCRIPTION
**Pull request by wingfeatherwave**
Screenshot 2023-08-30 at 10 04 31 AM

Test: 
* set local storage feature flag in browser console
`localStorage.setItem('feature:activity_contributions', true);`
* edit/delete one of your contribution on the circle activity page

### <samp>🤖 Generated by Copilot at 7a4449d</samp>

This pull request adds the ability to edit and delete contributions from the activity and contributions pages, using new hooks, components, and mutations. It also improves the test reliability, accessibility, and styling of some components, and fixes some data inconsistencies. The main files affected are `ContributionRow.tsx`, `ContributionForm.tsx`, `useInfiniteActivities.tsx`, and `ActivityRow.test.tsx`.

### <samp>🤖 Generated by Copilot at 7a4449d</samp>

> _The `ContributionForm` got a makeover_
> _With new hooks and props to pass over_
> _It can edit and delete_
> _With `ContributionRow` so neat_
> _And the `Button` component is easier to hover_

### <samp>🤖 Generated by Copilot at 7a4449d</samp>

*  Add editing and deleting functionality for contributions ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274L1-R9), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274R25-R28), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274L34-R43), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274L56-R97), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L1-R43), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L32-R82), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L56-R105), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L106-R223), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L132-R353), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-12ce896f4e03fe14468326cbb9302b08e814171c5f959115b739f14d22666cdfL22-R22))
  - Update `ContributionRow` component to show edit and delete buttons and toggle `ContributionForm` component ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274L1-R9), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274R25-R28), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274L34-R43), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-023fbc171b14533aea64ceea9c06efab32d5448fe6d51fd5f36dd43b7c711274L56-R97))
  - Update `ContributionForm` component to accept props and parameters for editing and deleting contributions, and to perform mutations and invalidate queries accordingly ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L1-R43), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L32-R82), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L56-R105), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L106-R223), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-8c417a5a2b551e89bf1c0521c1718e4fab41f3439b3b638f8aaa32f2ced22e60L132-R353))
  - Modify `getNewContribution` function in `util` file to handle edge case of no next contribution ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-12ce896f4e03fe14468326cbb9302b08e814171c5f959115b739f14d22666cdfL22-R22))
* Add `created_at` and `id` fields to contribution objects in queries and mock data ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-c193e252dd7b0cd1c9089baf58f4440b8024634baf0b4c1de6df3916d055b42eL126-R128), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-c193e252dd7b0cd1c9089baf58f4440b8024634baf0b4c1de6df3916d055b42eL168-R172), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-de71c4ba5014d75c0e22b56c2dea640e7a20fda5614d3928efcc786e8b663391R52-R53))
  - Update mock data for `ActivityRow` component test to match expected data shape ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-c193e252dd7b0cd1c9089baf58f4440b8024634baf0b4c1de6df3916d055b42eL126-R128), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-c193e252dd7b0cd1c9089baf58f4440b8024634baf0b4c1de6df3916d055b42eL168-R172))
  - Update `getActivities` query function to include `created_at` and `id` fields in contribution objects ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-de71c4ba5014d75c0e22b56c2dea640e7a20fda5614d3928efcc786e8b663391R52-R53))
* Pass `circleId` prop to `ContributionForm` component in `CircleActivityPage` component ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-ad4502b0dfe7c31d2c3282e694732d7e7e4d5095133d8274e63f50746492385dL28-R30))
* Add `focus-visible` styles to `Button` component variants for accessibility and keyboard navigation ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-02f9aee49ae60f1c5cc0946b7c193bca1b9570b3aef445dcc4f4ac9164af4394R20-R23), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-02f9aee49ae60f1c5cc0946b7c193bca1b9570b3aef445dcc4f4ac9164af4394R33-R35), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-02f9aee49ae60f1c5cc0946b7c193bca1b9570b3aef445dcc4f4ac9164af4394L37-R44), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-02f9aee49ae60f1c5cc0946b7c193bca1b9570b3aef445dcc4f4ac9164af4394L51-R57), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-02f9aee49ae60f1c5cc0946b7c193bca1b9570b3aef445dcc4f4ac9164af4394L73-R78), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-02f9aee49ae60f1c5cc0946b7c193bca1b9570b3aef445dcc4f4ac9164af4394L84-R88))
* Add margin and padding styles to `ul` and `ol` elements and remove extra space from last element in `StyledMarkdownPreview` component ([link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-d5d03ad513a3d46fed5483283ae5102d02393967d8ba4a3f54efc1090f0afa9dL27-R27), [link](https://github.com/coordinape/coordinape/pull/2305/files?diff=unified&w=0#diff-d5d03ad513a3d46fed5483283ae5102d02393967d8ba4a3f54efc1090f0afa9dR40-R44))

### CODE REVIEW

**Comment by wingfeatherwave** [comment]
updating copy/display:
Screenshot 2023-08-29 at 12 22 08 PM

File: src/features/activities/ContributionRow.tsx [hunk 1]
```diff
@@ -17,6 +22,10 @@ export const ContributionRow = ({
   drawer?: boolean;
 }) => {
   const { inCircle } = usePathContext();
+  const { data } = useNavQuery();
+  const editableContribution =
+    activity.actor_profile.address === data?.profile?.address;
```
Comment (crabsinger) [inline src/features/activities/ContributionRow.tsx:27]: i htink we might prefer doing this on profile.id vs address ? @CryptoGraffe ?

File: src/pages/ContributionsPage/ContributionForm.tsx [hunk 2]
```diff
@@ -103,16 +152,75 @@ export const ContributionForm = () => {
       },
     });
 
+  const { mutate: mutateContribution } = useMutation(
+    updateContributionMutation,
+    {
+      mutationKey: ['updateContribution', currentContribution?.contribution.id],
+      onError: (errors, { id }) => {
+        updateSaveStateForContribution(id, 'error');
```
Comment (crabsinger) [inline src/pages/ContributionsPage/ContributionForm.tsx:160]: maybe show a toast error on errors?

**Review by crabsinger — DISMISSED** [review]
looks good - a few renames and nits @wingfeatherwave

**Review by crabsinger — DISMISSED** [review]

**Review by crabsinger — APPROVED** [review]

**Pull request merged**